RIOT-OS/RIOT
# boards/stm32f4: remove default clock configuration for 96MHz and 168MHz CPUs

### Contribution description

In the STM32 F4 families CPUs can reach a maximum clock of either:
-  84MHz => stm32f401xe
- 100MHz => stm32f410xb, stm32f411xe, stm32f412, stm32f413
- 180MHz => all the others

In the current situation, f410 and f411 are all clocked by default at a maximum of 96MHz and f429, f405 at a maximum of 168MHz. The reason is that these families doesn't provide an alternative 48MHz clock other than the main PLL and it's not possible to configure PLLQ to 48MHz when PLLP is configured to output 100MHz or 180MHz (no common divisor).

This PR is modifying the PLL configuration as follows:
- 96MHz and 168MHz default max clock are only configured when needed, e.g. when periph USB dev is used
- otherwise, the maximum possible clock is configured




### Testing procedure

- Verify USB is still working on affected boards: nucleo-f429zi and on weact-f411re, stdio (which is provided on top of USB) should also work
- Verify that nucleo-f410rb, nucleo-f411re, nucleo-f429zi, stm32f4discovery, pyboard are still working. There max clock speed should be 100MHz (f410rb, f411re) and 180MHz now.




### Issues/PRs references

Based on #14946 
Tick an item in #14975
